### PR TITLE
ERM-1859

### PR DESCRIPTION
**Pull request by EthanFreestone**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
```
Comment (aditya-matukumalli) [inline src/index.js:1]: We arent using `useEffect` in this file?

File: src/routes/JobViewRoute.js [hunk 2]
```diff
@@ -8,104 +8,107 @@ import { ConfirmationModal } from '@folio/stripes/components';
...
+  }).isRequired,
+  resources: PropTypes.shape({
+    job: PropTypes.object,
+  }).isRequired,
+  stripes: PropTypes.shape({
+    okapi: PropTypes.object.isRequired,
+  }).isRequired,
+};
 
 export default stripesConnect(JobViewRoute);
```
Comment (aditya-matukumalli) [inline src/routes/JobViewRoute.js:114]: You could just use `useStripes` hook instead now that this is a fn component

File: src/routes/components/useExportLogStream.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  }, [callout, refetch]);
+
+  if (!job || !type) {
+    return {};
+  }
+
+  return { refetch: refetchWithCallout, isLoading };
+};
+
+export default useExportLogStream;
```
Comment (aditya-matukumalli) [inline src/routes/components/useExportLogStream.js:36]: Lets export this file from a hooks folder at the same level as the components folder

**Review by aditya-matukumalli — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (EthanFreestone) [reply src/index.js:1]: Whoops... I thought linting would have caught that? Good spot

Comment (EthanFreestone) [reply src/index.js:1]: Oh it did :p my bad

File: src/components/Logs/Logs.js [hunk 4]
```diff
@@ -1,32 +1,55 @@
 import React from 'react';
 import { FormattedMessage } from 'react-intl';
 import PropTypes from 'prop-types';
-import { Accordion, Badge } from '@folio/stripes/components';
+import { Accordion, Badge, Button, Spinner } from '@folio/stripes/components';
 
 import JobLogContainer from '../../containers/JobLogContainer';
+import useExportLogStream from '../../routes/components/useExportLogStream';
```
[the remark this answers is not in the document]
Comment (EthanFreestone) [reply src/components/Logs/Logs.js:7]: Dnoe

File: src/components/Logs/Logs.js [hunk 5]
```diff
@@ -1,32 +1,55 @@
...
+  type: PropTypes.string.isRequired,
+};
 
-  render() {
-    const { id, job, type } = this.props;
+const Logs = ({ id, job, type }) => {
+  const logCount = job[`${type}LogCount`];
 
+  const { refetch: fetchLogs, isLoading } = useExportLogStream(job, type);
+  const renderBadgeAndExport = () => {
```
[the remark this answers is not in the document]
Comment (EthanFreestone) [reply src/components/Logs/Logs.js]: Done

File: src/routes/JobViewRoute.js (hunk 2 above)
Comment (EthanFreestone) [reply src/routes/JobViewRoute.js:114]: Hmm, I actually don't see anywhere we use the stripes object anyway. We still need stripes connect for the manifest for the time being, but I could remove these from PropTypes?

File: src/routes/components/useExportLogStream.js (hunk 3 above)
Comment (EthanFreestone) [reply src/routes/components/useExportLogStream.js:36]: Done

**Review by aditya-matukumalli — APPROVED** [review]

**Pull request merged**